File: grocery store frontend/src/services/CartService.js
import axios from "axios";

const CART_BASE_URL = "http://localhost:8080/cart/"
class CartService{

    //Adding item to cart
    addToCart(cart)
    {
        return axios.post(CART_BASE_URL+"addToCart",cart);
    }

    //Get cartList of corresponding user
    getCartByUserId(userId)
    {
        console.log("CartService",userId);
        return axios.post(CART_BASE_URL+"getCartList/"+userId);
    }

    //Editing Cart
    editCart(id,cart)
    {
        return axios.put(CART_BASE_URL+"editCart/"+id,cart);
    }

    //Deleting Cart

    deleteCart(id)
    {
        return axios.delete(CART_BASE_URL+"delete/"+id);
    }

}

export default new CartService();

File: grocery store backend/src/main/java/com/strafe/service/CartService.java
package com.strafe.service;

import com.strafe.entity.CartEntity;

import java.util.List;

public interface CartService {
    CartEntity addToCart(CartEntity cart);

    String deleteCart(long id);

    CartEntity editCart(long id, CartEntity cart);

    List<CartEntity> getCartListByUserId(long userId);

    String deleteCartByUserId(Long userId);
}
